tests/ftqc/test_decomposition.py

```diff
@@ -192,3 +194,235 @@ def rot_mbqc(rot_xzx_gate):
         xzx_op = q.queue[0]
 
         assert np.allclose(rot_ref(angles), rot_mbqc(xzx_op))
+
+
+class TestMBQCFormalismConversion:
+    """Test the transform convert_to_mbqc_formalism, converting to the MBQC formalism with
+    online corrections immediately after each gate"""
+
+    @pytest.mark.parametrize(
+        "op",
+        [qml.H(2), qml.S(2), qml.RZ(1.23, 2), RotXZX(0, 1.23, 0, 2), RotXZX(0.12, 0.34, 0.56, 2)],
+    )
+    def test_convert_single_qubit_gate(self, op):
+        dev = qml.device("lightning.qubit", wires=5)
+        q_mgr = QubitMgr(num_qubits=5, start_idx=0)
+        wire_map = {2: q_mgr.acquire_qubit()}
+        w = op.wires[0]
+
+        ref_tape = qml.tape.QuantumScript(
+            [qml.Rot(1.2, 0.34, 0.7, w), op],
+            measurements=[qml.expval(qml.X(w)), qml.expval(qml.Y(w)), qml.expval(qml.Z(w))],
+        )
+
+        with qml.queuing.AnnotatedQueue() as q:
+            qml.Rot(1.2, 0.34, 0.7, wire_map[w])
+            wire_map[w] = queue_single_qubit_gate(q_mgr, op=op, in_wire=wire_map[w])
+            qml.expval(qml.X(wire_map[w]))
+            qml.expval(qml.Y(wire_map[w]))
+            qml.expval(qml.Z(wire_map[w]))
+
+        tape = qml.tape.QuantumScript.from_queue(q, shots=3000)
+        (diagonalized_tape,), _ = diagonalize_mcms(tape)
+
+        res, res_ref = qml.execute([diagonalized_tape, ref_tape], device=dev, mcm_method="one-shot")
+        assert np.allclose(res, res_ref, atol=0.05)
+
+    def test_convert_cnot(self):
+
+        dev = qml.device("lightning.qubit", wires=15)
```

Comment:
Any concerns with runtime of this test?

Reply:
What is the cutoff where we become concerned? Locally this test is 8.5 seconds for me. The slowest test here is `test_conversion_of_multi_qubit_circuit` below, which takes 15 seconds on my computer.

Reply:
I remember we do have a marker for those who are potentially slow: `pytest.mark.slow`; we can try it if runtime becomes a concern

Reply:
A few seconds is fine by me. I'd say 10s+ is something to mark as slow, but I'd be ok with 8s for now.

Reply:
Marked the 15 second test as slow 🐢